### PR TITLE
mpvScripts.videoclip: init at 0-unstable-2024-03-08

### PR DESCRIPTION
**Pull request by BatteredBunny**
## Description of changes

Add [videoclip](https://github.com/Ajatt-Tools/videoclip)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/videoclip.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
+, stdenvNoCC
+, curl
+, xclip
+, wl-clipboard
+, stdenv
+,
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/videoclip.nix:8]:
```suggestion
```

File: pkgs/applications/video/mpv/scripts/videoclip.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  dontBuild = true;
+
+  patchPhase =
+    ''
+      substituteInPlace platform.lua \
+      --replace \'curl\' \'${lib.getExe curl}\' \
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      --replace xclip ${lib.getExe xclip} \
+      --replace wl-copy ${lib.getBin wl-clipboard}/bin/wl-copy
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/videoclip.nix:30]:
```suggestion
      --replace wl-copy ${lib.getExe' wl-clipboard "wl-copy"}
```

Comment (eclairevoyant) [reply pkgs/applications/video/mpv/scripts/videoclip.nix:30]: also please keep the pre/post patch hooks

File: pkgs/applications/video/mpv/scripts/videoclip.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    runHook postInstall
+  '';
+
+  passthru.scriptName = "videoclip";
+
+  meta = with lib; {
+    description = "Easily create videoclips with mpv.";
+    homepage = "https://github.com/Ajatt-Tools/videoclip";
+    license = licenses.gpl3;
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/videoclip.nix:47]: please use a valid identifier (gpl3Only or gpl3Plus as per upstream)

File: pkgs/applications/video/mpv/scripts/videoclip.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    mkdir -p $out/share/mpv/scripts/videoclip
+    cp *.lua $out/share/mpv/scripts/videoclip
+
+    runHook postInstall
+  '';
+
+  passthru.scriptName = "videoclip";
+
+  meta = with lib; {
+    description = "Easily create videoclips with mpv.";
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/videoclip.nix:45]:
```suggestion
    description = "Easily create videoclips with mpv";
```

never end desc with a period as per nixpkgs manual

File: pkgs/applications/video/mpv/scripts/videoclip.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    homepage = "https://github.com/Ajatt-Tools/videoclip";
+    license = licenses.gpl3;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ ayes-web ];
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/videoclip.nix:49]: are you added to the maintainers list? if not please do so in the first commit, then init this paxkage in the second commit

**Review by eclairevoyant — DISMISSED** [review]

**Review by nicoonoclaste — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution!  :purple_heart: 

This could likely be simplified using [`buildLua`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/video/mpv/scripts/buildLua.nix), a build helper in `mpvScripts`

**Review by nicoonoclaste — APPROVED** [review]
Just an errant colon to clean up, which I can do myself  ;3

**Pull request merged**